app/core/tests/test_commands.py

```diff
@@ -0,0 +1,23 @@
+from unittest.mock import patch
+
+from django.core.management import call_command
+from django.db.utils import OperationalError
+from django.test import TestCase
+
+
+class CommandTests(TestCase):
+
+    def test_wait_for_db_ready(self):
+        """test waiting for db"""
+        with patch('django.db.utils.ConnectionHandler.__getitem__') as gi:
+            gi.return_value = True
+            call_command('wait_for_db')
+            self.assertEqual(gi.call_count, 1)
+
+    @patch('time.sleep', return_value=True)
+    def test_wait_for_db(self, ts):
+        """wait for db"""
+        with patch('django.db.utils.ConnectionHandler.__getitem__') as gi:
```

Comment:
Out of curiosity, why do you use both `patch` options, decorator and context manager?

Reply:
I agree this sttyle is confusing, I followed the course where they explained that `patch` can be used in different ways. I would prefer to write a full version 2 times. About logic - this is to remove sleep on db reconnection for tests, otherwise it waits 5 seconds just for this test.